File: plsql-squid/src/test/java/br/com/felipezorzo/sonar/plsql/api/units/AnonymousBlockTest.java
package br.com.felipezorzo.sonar.plsql.api.units;

import static org.sonar.sslr.tests.Assertions.assertThat;

import org.junit.Before;
import org.junit.Test;

import br.com.felipezorzo.sonar.plsql.api.PlSqlGrammar;
import br.com.felipezorzo.sonar.plsql.api.RuleTest;

public class AnonymousBlockTest extends RuleTest {

    @Before
    public void init() {
        setRootRule(PlSqlGrammar.ANONYMOUS_BLOCK);
    }

    @Test
    public void matchesSimpleBlock() {
        assertThat(p).matches(""
                + "BEGIN\n"
                + "NULL;\n"
                + "END;");
    }
    
    @Test
    public void matchesBlockWithDeclareSection() {
        assertThat(p).matches(""
                + "DECLARE\n"
                + "VAR NUMBER;\n"
                + "BEGIN\n"
                + "NULL;\n"
                + "END;");
    }

}

File: zpa-core/src/test/kotlin/org/sonar/plugins/plsqlopen/api/units/AnonymousBlockTest.kt
/**
 * Z PL/SQL Analyzer
 * Copyright (C) 2015-2022 Felipe Zorzo
 * mailto:felipe AT felipezorzo DOT com DOT br
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
package org.sonar.plugins.plsqlopen.api.units

import com.felipebz.flr.tests.Assertions.assertThat
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.sonar.plugins.plsqlopen.api.PlSqlGrammar
import org.sonar.plugins.plsqlopen.api.RuleTest

class AnonymousBlockTest : RuleTest() {

    @BeforeEach
    fun init() {
        setRootRule(PlSqlGrammar.ANONYMOUS_BLOCK)
    }

    @Test
    fun matchesSimpleBlock() {
        assertThat(p).matches(""
                + "BEGIN\n"
                + "NULL;\n"
                + "END;")
    }

    @Test
    fun matchesBlockWithDeclareSection() {
        assertThat(p).matches(""
                + "DECLARE\n"
                + "VAR NUMBER;\n"
                + "BEGIN\n"
                + "NULL;\n"
                + "END;")
    }

}
